src/main/java/io/pakland/mdas/githubstats/domain/repository/CommentExternalRepository.java

```diff
@@ -2,26 +2,11 @@
 
 import io.pakland.mdas.githubstats.application.exceptions.HttpException;
 import io.pakland.mdas.githubstats.domain.entity.Comment;
-import lombok.AllArgsConstructor;
-import lombok.Builder;
-import lombok.Getter;
-import lombok.NoArgsConstructor;
+import io.pakland.mdas.githubstats.domain.entity.PullRequest;
 
 import java.util.List;
 
 public interface CommentExternalRepository {
 
-    public List<Comment> fetchCommentsFromPullRequest(FetchCommentsFromPullRequestRequest request) throws HttpException;
-
-    @NoArgsConstructor
-    @AllArgsConstructor
-    @Builder
-    @Getter
-    public static class FetchCommentsFromPullRequestRequest {
-        private String repositoryOwner;
-        private String repositoryName;
-        private Integer pullRequestNumber;
-        private Integer page;
-        private Integer perPage;
-    }
+    public List<Comment> fetchCommentsFromPullRequestByPage(PullRequest pullRequest, Integer page) throws HttpException;
```

Comment:
This can be resumed as removing the request class and just passing the entity that is required?

Reply:
I think that the page parameter should be appart from the entity parameter as it belongs to the filtering we do on this specific repository request. I've used entities as they are not dependant on the needs of the Github API, so if GitLab requires another parameter that we hold on the PullRequest entity we can still implement its external repository adapter.